Assets/Scripts/Enviroment/EnvironmentBehaviour.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnvironmentBehaviour : MonoBehaviour
+{
+    [Header("Stats")]
+    public float moveSpeed;
+    public float lifeSpan;
+
+    [Header("Despawn when Touching:")] 
+    public bool tree;
+    public bool bush;
+    public bool rock;
+    public bool squiggle;
+    public bool water;
+    
+    private float age;
+    
+    // Update is called once per frame
+    void Update()
+    {
+        Move();
+        Despawn();
+    }
+
+    void Move()
+    {
+        transform.position += Vector3.down * moveSpeed;
+    }
+    
+    void Despawn()
+    {
+        age += Time.deltaTime;
+
+        if (age >= lifeSpan)
+        {
+            Destroy(gameObject);
+        }
+    }
+    
+    private void OnTriggerEnter2D(Collider2D col)
+    {
+        if (CompareTag("Tree") && col.CompareTag("Tree") && tree)
+        {
+            if (age <= 0.5f)
+            {
+                Destroy(gameObject);
+            }
```

Comment:
Waarom is het alleen bij tree dat je een age heb?

Reply:
oh shit die was nog niet af